src/Analyzers/CSharp/Analyzers/RemoveRedundantElseStatement/RemoveRedundantElseStatementDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,136 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+using Microsoft.CodeAnalysis.CodeStyle;
+using Microsoft.CodeAnalysis.CSharp.CodeStyle;
+using Microsoft.CodeAnalysis.CSharp.Extensions;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Operations;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+
+namespace Microsoft.CodeAnalysis.CSharp.RemoveRedundantElseStatement
+{
+    /// <summary>
+    /// Looks for code like:
+    ///
+    ///     ```c#
+    ///     if (a == b)
+    ///     {
+    ///         return c;
+    ///     }
+    ///     else
+    ///     {
+    ///         return d;
+    ///     }
+    ///     ```
+    /// 
+    /// And offers to convert it to:
+    ///
+    ///     ```c#
+    ///     if (a == b)
+    ///     {
+    ///         return c;
+    ///     }
+    ///     
+    ///     return d;
+    ///     ```
+    ///
+    /// For this conversion to make sense statement(s) in each `if` and `else if`
+    /// must end with a jump (return, break, continue or throw) 
+    /// in order to preserve the program's correctness.
+    /// </summary>
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    internal class RemoveRedundantElseStatementDiagnosticAnalyzer : AbstractBuiltInCodeStyleDiagnosticAnalyzer
```

Comment:
```suggestion
    internal sealed class RemoveRedundantElseStatementDiagnosticAnalyzer : AbstractBuiltInCodeStyleDiagnosticAnalyzer
```

Reply:
somewhat surprised this isn't deriving from a common type in the shared layer.  That will be helpful for sharing code with VB impl.